lib/sensors/accelerometer.hpp

```diff
@@ -48,14 +50,17 @@ constexpr std::uint8_t kDataReady = 0x27;
 constexpr std::uint8_t kDeviceIdAddress       = 0x0F;
 constexpr std::uint8_t kExpectedDeviceIdValue = 0x44;
 
-class Accelerometer : public II2cMuxSensor<core::RawAccelerationData> {
+class Accelerometer : public II2cMuxSensor<core::RawAcceleration> {
  public:
-  Accelerometer(core::ILogger &logger, io::II2c &i2c, const std::uint8_t channel);
+  Accelerometer(core::ILogger &logger,
+                core::ITimeSource &time_source,
+                io::II2c &i2c,
+                const std::uint8_t channel);
   ~Accelerometer();
 
-  core::Result configure();
-  std::optional<core::RawAccelerationData> read();
-  std::uint8_t getChannel() const;
+  virtual core::Result configure();
+  virtual std::optional<core::Measurement<core::RawAcceleration>> read();
+  virtual std::uint8_t getChannel() const;
```

Comment:
i think there are plenty methods currently in derived classes that omit `virtual`. dont thinkit changes anything if we omit it or not? but reckon we should decide on the standard, im okay with adding this to derived method from now on.

Reply:
I would say that we should definitely include virtual. It just makes it clearer that the signature comes from the interface.

Reply:
cool, will start enforcing this